packages/client/src/runtime/core/engines/library/LibraryEngine.ts

```
@@ -81,36 +81,30 @@ export class LibraryEngine extends Engine<undefined> {
     version: string
   }
 
-  constructor(config: EngineConfig, loader: LibraryLoader = new DefaultLibraryLoader(config)) {
+  constructor(config: EngineConfig, libraryLoader?: LibraryLoader) {
     super()
 
-    try {
-      // we try to handle the case where the datamodel is not found
-      this.datamodel = fs.readFileSync(config.datamodelPath, 'utf-8')
-    } catch (e) {
-      if ((e.stack as string).match(/\/\.next|\/next@|\/next\//)) {
-        throw new PrismaClientInitializationError(
-          `Your schema.prisma could not be found, and we detected that you are using Next.js.
-Find out why and learn how to fix this: https://pris.ly/d/schema-not-found-nextjs`,
-          config.clientVersion!,
-        )
-      } else if (config.isBundled === true) {
-        throw new PrismaClientInitializationError(
-          `Prisma Client could not find its \`schema.prisma\`. This is likely caused by a bundling step, which leads to \`schema.prisma\` not being copied near the resulting bundle. We would appreciate if you could take the time to share some information with us.
-Please help us by answering a few questions: https://pris.ly/bundler-investigation-error`,
-          config.clientVersion!,
-        )
-      }
+    const engineType = getClientEngineType(config.generator!)
 
-      throw e
+    if (TARGET_BUILD_TYPE === 'library') {
+      // for "library" builds, we can use both the wasm and native engines
+      if (engineType === ClientEngineType.Wasm && config.adapter !== undefined) {
+        this.libraryLoader = libraryLoader ?? wasmLibraryLoader
+      } else {
+        this.libraryLoader = libraryLoader ?? defaultLibraryLoader
```

Should we throw if `engineType` is WASM, but adapter is not set?